src/tiered/tiered_work.c

```diff
@@ -183,7 +183,7 @@ __wt_tiered_flush_work_wait(WT_SESSION_IMPL *session, uint32_t timeout)
     conn = S2C(session);
     __wt_epoch(session, &start);
     now = start;
-    done = found = false;
+    done = false;
```

Comment:
This is an unfortunate consequence of our C coding standard. As `found` is only actually used within the `while` loop, it would be better to be able to declare `found` there rather than at the beginning of the function. 
The proposed change to remove the initialisation of `found`, while removing a warning, raises the risk of the variable becoming uninitialised if someone makes a mistake in changing the code later on.

Reply:
Yes but compilers will warn if the variable could be used uninitialized so I'd trust the compiler here instead of relying on a dead store.

Reply:
Sadly, there are cases the compiler cannot detect. For example, passing a pointer to a local variable into a function that is in another .c file where the compiler can't know if the value is initialised by that function or not.